packages/nimble-components/src/anchor/index.ts

```diff
@@ -47,6 +48,10 @@ export class Anchor extends AnchorBase {
      */
     @attr({ attribute: 'contenteditable' })
     public override contentEditable!: string;
+
+    /** @internal */
+    @observable
+    public tabIndexOverride = 0;
```

Comment:
Elements in the table that are focusable need a way for the table to override that behavior, for the roving tabindex approach to work. (This is for controls with `tabindex=0` explicitly on the control or something in its template, or that are implicitly focusable like `<a>`)

The current approach for this is:
- Add a `tabIndexOverride` property for those elements (defaults to `0` so that non-table uses work the same as they do now)
- In the element template, where `tabindex=0` was set previously, use `tabindex=tabIndexOverride` instead. (Note: Here in the anchor tabindex wasn't set explicitly before, but tabindex=0 is the default for `<a>`)
- In the templates inside the table, set `tabIndexOverride` to `-1` (i.e. not keyboard focusable by default)
- `KeyboardNavigationManager` will set the current focus in the table to `tabIndex`/`tabIndexOverride`=`0` explicitly before focusing it, and back to `-1` when it's unfocused.

Reply:
I don't think I see from this description why we need a different name for tabindex. Fundamentally we want the components to match the native APIs as closely as possible. Why can't we implement the tabindex api correctly?

Reply:
At the time I'd added this, I'd run into issues trying to set `tabIndex` directly for these controls, due to them already setting `tabIndex` in their templates. I can take another stab at getting it working though.

Matching [what we do for the AnchorMenuItem](https://github.com/ni/nimble/blob/c1c3f73d50d327368f8aba56206208e93c22cfd1/packages/nimble-components/src/anchor-menu-item/index.ts#L73) (overriding `setAttribute()` and `get/set tabIndex`) didn't seem to be sufficient.

Reply:
Created a separate work item for the tabIndex work: #2094

Reply:
Also removed all the tabIndexOverride code from this PR. Replaced with (temporary) `NotFocusableBehavior` and `getTabIndexTarget()`.